pkg/controller/aws/network/subnet/controller.go

```diff
@@ -0,0 +1,191 @@
+)
+
+// Controller is the controller for Subnet objects
+type Controller struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha1.SubnetGroupVersionKind),
+		resource.WithExternalConnecter(&connector{client: mgr.GetClient(), newClientFn: ec2.NewSubnetClient, awsConfigFn: utils.RetrieveAwsConfigFromProvider}),
+		resource.WithManagedConnectionPublishers())
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha1.SubnetKindAPIVersion, v1alpha1.Group))
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Subnet{}).
+		Complete(r)
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) (ec2.SubnetClient, error)
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha1.Subnet)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := conn.newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create a new SubnetClient")
+	}
+
+	return &external{c}, nil
+}
+
+type external struct {
+	client ec2.SubnetClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.Subnet)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	// To find out whether a Subnet exist:
+	// - the object's ExternalState should have subnetId populated
+	// - a Subnet with the given subnetId should exist
+	if cr.Status.SubnetID == "" {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	req := e.client.DescribeSubnetsRequest(&awsec2.DescribeSubnetsInput{
+		SubnetIds: []string{cr.Status.SubnetID},
+	})
+	req.SetContext(ctx)
+
+	response, err := req.Send()
+
+	if err != nil {
+		if ec2.IsSubnetNotFoundErr(err) {
+			return resource.ExternalObservation{
+				ResourceExists: false,
+			}, nil
+		}
+
+		return resource.ExternalObservation{}, errors.Wrapf(err, "failed to describe Subnet with id: %v", cr.Status.SubnetID)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.Subnets) != 1 {
+		return resource.ExternalObservation{}, errors.Errorf("retrieved multiple Subnet for the given subnetId: %v", cr.Status.SubnetID)
+	}
+
+	observed := response.Subnets[0]
+
+	if observed.State == awsec2.SubnetStateAvailable {
+		cr.SetConditions(runtimev1alpha1.Available())
```

Comment:
https://godoc.org/github.com/aws/aws-sdk-go-v2/service/ec2#SubnetState

Does `pending` correspond to `creating?` Perhaps we should handle that case here.